core/time/timezone/tz_unix.odin

```diff
@@ -0,0 +1,85 @@
+#+build darwin, linux, freebsd, openbsd, netbsd
+#+private
+package timezone
+
+import "core:os"
+import "core:strings"
+import "core:path/filepath"
+import "core:time/datetime"
+
+local_tz_name :: proc(allocator := context.allocator) -> (name: string, success: bool) {
+	local_str, ok := os.lookup_env("TZ", allocator)
+	if !ok {
+		orig_localtime_path := "/etc/localtime"
+		path, err := os.absolute_path_from_relative(orig_localtime_path)
```

Comment:
This call allocates (implicitly because the os package is bad) so you'll have to set `context.allocator = allocator` before calling it.
Or you could change the proc in the os package to take an allocator.